fix: validate files to be uploaded

trevorblades opened the pull request:
This PR is part one of two, ensuring that only files of specific types are uploaded to our system. This does file upload validation on the client side, but we must also implement some server-side validation.

Fixes FER-2070

dsinghvi reviewed (commented):
@trevorblades thoughts on adding this to the `docs-validation` package ([link](https://github.com/fern-api/fern/tree/main/packages/cli/yaml/docs-validator/src/rules))? 

This way when a user runs `fern check` they will get a validation error that will automatically notify them that they can't upload those specific files.

trevorblades commented:
> @trevorblades thoughts on adding this to the `docs-validation` package ([link](https://github.com/fern-api/fern/tree/main/packages/cli/yaml/docs-validator/src/rules))?
> 
> This way when a user runs `fern check` they will get a validation error that will automatically notify them that they can't upload those specific files.

That's a good idea! I can do that. Does the docs validator run before docs publishing, and would its failure stop a publish command? What I'm getting at is if you see this as something that lives in the docs validator _and_ here in the publish command, or just in the validator.

dsinghvi commented:
> > @trevorblades thoughts on adding this to the `docs-validation` package ([link](https://github.com/fern-api/fern/tree/main/packages/cli/yaml/docs-validator/src/rules))?
> > This way when a user runs `fern check` they will get a validation error that will automatically notify them that they can't upload those specific files.
> 
> That's a good idea! I can do that. Does the docs validator run before docs publishing, and would its failure stop a publish command? What I'm getting at is if you see this as something that lives in the docs validator _and_ here in the publish command, or just in the validator.

Just in the validator works! the validator runs and has to be passed in order to hit publish

trevorblades commented:
@dsinghvi I moved the file type check to the validator CLI and added some tests to confirm it's working as expected. Lmk if that looks like what you had in mind!

dsinghvi reviewed (approved):
🚀 thank you!

abvthecity commented on packages/cli/yaml/docs-validator/src/rules/valid-file-types/valid-file-types.ts:
```diff
@@ -27,16 +28,20 @@ export const ValidFileTypes: Rule = {
+                            }
+                        ];
+                    }
                 }
```
should we add a warning or error if the file doesn't exist?

dsinghvi replied:
there's another rule that checks that!
